Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1496,17 +1510,19 @@ PhysicalParticleContainer::AddPlasmaFlux (amrex::Real dt)
                 const XDim3 r =
                     inj_pos->getPositionUnitBox(i_part, lrrfac, engine);
                 auto pos = getCellCoords(overlap_corner, dx, r, iv);
+                auto ppos = PDim3(pos);
 
                 // inj_mom would typically be InjectorMomentumGaussianFlux
                 XDim3 u;
                 u = inj_mom->getMomentum(pos.x, pos.y, pos.z, engine);
+                auto pu = PDim3(u);
 
                 u.x *= PhysConst::c;
                 u.y *= PhysConst::c;
                 u.z *= PhysConst::c;
 
                 const amrex::Real t_fract = amrex::Random(engine)*dt;
-                UpdatePosition(pos.x, pos.y, pos.z, u.x, u.y, u.z, t_fract);
+                UpdatePosition(ppos.x, ppos.y, ppos.z, pu.x, pu.y, pu.z, t_fract);
```

Comment:
The change to add and use `PDim3` is incomplete. The values in `ppos` and `pu` are going to be modified by this function, but calls below still use `pos`

Reply:
Good catch, I'll update the code below.